libnavui-androidauto/src/test/java/com/mapbox/androidauto/car/MapboxScreenManagerTest.kt

```diff
@@ -0,0 +1,221 @@
+package com.mapbox.androidauto.car
+
+import androidx.car.app.CarContext
+import androidx.car.app.Screen
+import androidx.car.app.ScreenManager
+import com.mapbox.android.core.permissions.PermissionsManager
+import com.mapbox.androidauto.ActiveGuidanceState
+import com.mapbox.androidauto.ArrivalState
+import com.mapbox.androidauto.CarAppState
+import com.mapbox.androidauto.FreeDriveState
+import com.mapbox.androidauto.MapboxCarApp
+import com.mapbox.androidauto.RoutePreviewState
+import com.mapbox.androidauto.car.feedback.ui.CarGridFeedbackScreen
+import com.mapbox.androidauto.car.navigation.ActiveGuidanceScreen
+import com.mapbox.androidauto.car.permissions.NeedsLocationPermissionsScreen
+import com.mapbox.androidauto.car.preview.CarRoutePreviewScreen
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.testing.MainCoroutineRule
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import io.mockk.verify
+import io.mockk.verifyOrder
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(
+    MapboxExperimental::class,
+    ExperimentalPreviewMapboxNavigationAPI::class,
+    ExperimentalCoroutinesApi::class
+)
+class MapboxScreenManagerTest {
+
+    @get:Rule
+    var coroutineRule = MainCoroutineRule()
+
+    private val screenManager: ScreenManager = mockk(relaxed = true)
+    private val mockCarContext: CarContext = mockk {
+        every { getCarService(ScreenManager::class.java) } returns screenManager
+    }
+    private val mainCarContext: MainCarContext = mockk {
+        every { carContext } returns mockCarContext
+    }
+    private val screenProvider: MapboxScreenProvider = mockk(relaxed = true)
+
+    private val sut = MapboxScreenManager(mainCarContext, screenProvider)
+
+    @Before
+    fun setup() {
+        mockkStatic(PermissionsManager::class)
+        mockkObject(MapboxCarApp)
+    }
+
+    @After
+    fun teardown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun `currentScreen is NeedsLocationPermissionsScreen permissions are not accepted`() {
+        every { PermissionsManager.areLocationPermissionsGranted(any()) } returns false
+        every {
+            screenProvider.needsLocationPermission()
+        } returns mockk<NeedsLocationPermissionsScreen>()
+
+        val screen = sut.currentScreen()
+
+        assertTrue(screen is NeedsLocationPermissionsScreen)
```

Comment:
This will produce a more meaningful message in case of a test failure:
```suggestion
        assertThat(screen, instanceOf(NeedsLocationPermissionsScreen::class.java))
```
The same applies for the rest of the tests.